portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -126,4 +143,42 @@ private static String convertToJsonUsingGson(ArrayList<Comment> list) {
         Gson gson = new Gson();
         return gson.toJson(list);
     }
+
+    /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+    private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+        BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+        Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+        List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+        // User submitted form without selecting a file, so we can't get a URL. (dev server)
```

Comment:
What is `(dev server)`?

Reply:
I'm guessing there is a different way of getting the URL between the test server and a deployed server: I'm not sure how it works but the example had it this way so I assumed it was important to have it this way